plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1264,6 +1332,127 @@ private List<ColumnMetadata> getColumnMetadatas(Table table)
+                handle.getSnapshotId(),
+                handle.getTableSchemaJson(),
+                handle.getPartitionSpecJson(),
+                handle.getFormatVersion(),
+                handle.getNameMappingJson(),
+                handle.getTableLocation(),
+                handle.getStorageProperties(),
+                analyzeColumnNames);
+    }
+
+    @Override
+    public TableStatisticsMetadata getStatisticsCollectionMetadata(ConnectorSession session, ConnectorTableMetadata tableMetadata)
+    {
+        Optional<Set<String>> analyzeColumns = getAnalyzeColumns(tableMetadata.getProperties());
+        Set<ColumnStatisticMetadata> columnStatistics = tableMetadata.getColumns().stream()
+                .filter(column -> !column.isHidden())
+                .filter(column -> analyzeColumns.isEmpty() || analyzeColumns.get().contains(column.getName()))
+                // TODO: add support for NDV summary/sketch, but using Theta sketch, not HLL; see https://github.com/apache/iceberg-docs/pull/69
+                .map(column -> new ColumnStatisticMetadata(column.getName(), NUMBER_OF_DISTINCT_VALUES))
+                .collect(toImmutableSet());
+        return new TableStatisticsMetadata(columnStatistics, ImmutableSet.of(), ImmutableList.of());
+    }
+
+    @Override
+    public ConnectorTableHandle beginStatisticsCollection(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        IcebergAnalyzeHandle handle = (IcebergAnalyzeHandle) tableHandle;
+        if (handle.getFormatVersion() < 2) {
+            throw new TrinoException(NOT_SUPPORTED, "Iceberg table statistics updates require at least format version 2");
+        }
+
+        verify(transaction == null, "transaction already set");
+        transaction = catalog.loadTable(session, handle.getSchemaTableName()).newTransaction();
+        return handle;
+    }
+
+    @Override
+    public void finishStatisticsCollection(ConnectorSession session, ConnectorTableHandle tableHandle, Collection<ComputedStatistics> computedStatistics)
+    {
+        IcebergAnalyzeHandle handle = (IcebergAnalyzeHandle) tableHandle;
+
+        if (computedStatistics.isEmpty()) {
+            // No statistics collected
+            transaction = null;
+            return;
+        }
+
+        UpdateTableStatistics updateStatistics = transaction.newUpdateTableStatistics();
+        Map<String, Integer> columnNameToId = transaction.table().schema().columns().stream()
+                .collect(toImmutableMap(Types.NestedField::name, Types.NestedField::fieldId));
+        StatisticsFile statisticsFile = statisticsFileWriter.writeStatistics(
+                session,
+                handle,
+                columnNameToId,
+                transaction.table().currentSnapshot().sequenceNumber(),
+                computedStatistics);
+        if (transaction.table().currentSnapshot() != null && transaction.table().currentSnapshot().statisticsFiles() != null) {
```

Comment:
should `transaction.table().currentSnapshot().statisticsFiles()` return null? I would expect empty list if not present.

Reply:
i agree but see my response to @zinking similar comment: https://github.com/apache/iceberg/pull/4741#discussion_r872142932